src/sql/base/browser/ui/infoBox/infoBox.ts

```diff
@@ -0,0 +1,102 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import 'vs/css!./media/infoBox';
+import { Disposable } from 'vs/base/common/lifecycle';
+import { status } from 'vs/base/browser/ui/aria/aria';
+import { IThemable } from 'vs/base/common/styler';
+import { Color } from 'vs/base/common/color';
+
+export interface IInfoBoxStyles {
+	informationBackground?: Color;
+	warningBackground?: Color;
+	errorBackground?: Color;
+	successBackground?: Color;
+}
+
+export type InfoBoxStyle = 'information' | 'warning' | 'error' | 'success';
+
+export interface InfoBoxOptions {
+	text: string;
+	style: InfoBoxStyle;
+}
+
+export class InfoBox extends Disposable implements IThemable {
+	private _imageElement: HTMLDivElement;
+	private _textElement: HTMLDivElement;
+	private _infoBoxElement: HTMLDivElement;
+	private _text = '';
+	private _infoBoxStyle: InfoBoxStyle = 'information';
+	private _styles: IInfoBoxStyles;
+
+	constructor(container: HTMLElement, options?: InfoBoxOptions) {
+		super();
+		this._infoBoxElement = document.createElement('div');
+		this._imageElement = document.createElement('div');
+		this._imageElement.setAttribute('role', 'image');
+		this._textElement = document.createElement('div');
+		this._textElement.classList.add('infobox-text');
+		container.appendChild(this._infoBoxElement);
+		this._infoBoxElement.appendChild(this._imageElement);
+		this._infoBoxElement.appendChild(this._textElement);
+		if (options) {
+			this.infoBoxStyle = options.style;
+			this.text = options.text;
+		}
+	}
+
+	public style(styles: IInfoBoxStyles): void {
+		this._styles = styles;
+		this.updateStyle();
+	}
+
+
+	public get infoBoxStyle(): InfoBoxStyle {
+		return this._infoBoxStyle;
+	}
+
+	public set infoBoxStyle(style: InfoBoxStyle) {
+		this._infoBoxStyle = style;
+		this._infoBoxElement.classList.remove(...this._infoBoxElement.classList);
+		this._imageElement.classList.remove(...this._imageElement.classList);
+		this._imageElement.setAttribute('aria-label', style);
+		this._infoBoxElement.classList.add('infobox-container', style);
+		this._imageElement.classList.add('infobox-image', style);
+		this.updateStyle();
+	}
+
+	public get text(): string {
+		return this._text;
+	}
+
+	public set text(text: string) {
+		if (this._text !== text) {
+			status(text);
```

Comment:
Hm, what's the use case for this thing? I was under the impression it was just a fancy text box with an icon - in which case do we really want to have it read out the content each time we set the text?

Reply:
this can be a static box as well as a notification, in case of migration wizard, the content will be updated when a button is clicked. to make this accessible, I added this.

Reply:
mmmm, but if it's a static box do we want that functionality? We don't read out text anywhere else we update it so that seems odd. Is this what the portal does for these info boxes?

Reply:
it does announce the changes, I tested it using create pool experience, the infobox shown below was announced when it showed up.
![image](https://user-images.githubusercontent.com/13777222/105557755-d2340480-5cc1-11eb-9a5a-cfbd2fd4b1fb.png)

Reply:
So that makes sense if it's supposed to be a "pop in and display something" feature. I guess I just saw it as more generic box to display information in with a type symbol next to it. 

I agree that having it announce if it's added to the DOM like that makes sense. But then I guess I have to ask why are we using this over just the toast feature which we already have built in? It's basically just a toast for a dialog the way it's being used...

Reply:
Eh, thinking about it I guess it'd be a toast if it was like an error message. But in your screenshot above using a toast for that would be kind of odd.

So all right - as far as what your use case is going to be this seems ok. I can see this being more generically useful in the future for just adding "fancy text" in a box (that doesn't change/is always shown) that shouldn't be announced when being added to the page. But we can add that functionality in to optionally not announce if we ever need that.

I would strongly suggest though that you make sure to call out in the doc comment in the API that it will announce the text when the text changes to make sure people are clear what the behavior is.

Reply:
@aasimkhan30 could you please send Charles the link to the mockup, I couldn't find it now.